src/shipit_code_coverage/shipit_code_coverage/codecov.py

```diff
@@ -49,7 +50,11 @@ def download_coverage_artifacts(build_task_id):
 
 def get_github_commit(mercurial_commit):
     url = 'https://api.pub.build.mozilla.org/mapper/gecko-dev/rev/hg/%s'
-    r = requests.get(url % mercurial_commit)
+    while True:
```

Comment:
Just to be clear I'm not oposing ``while True:`` I'm just thinking how to avoid mapper, since it is a bottleneck here. 

Another option could also be to generate an html coverage report and push it to some s3 bucket and have data for all commits there.